### ISSUE TITLE
Correlations: Improve accessibility

### ISSUE THREAD

**Issue by ifrost** [grafana/grafana#75868]
__Background__

As part of releasing Correlations, we performed a11y review. [The report](https://docs.google.com/document/d/1PagfYDwRum_cAcG2NAfS-3m0siVGdgb4Kw6ENaglgNg/edit) didn't identify major blockers, though there's a few items that should/could be addressed to improve a11y:

Feel free to tackle multiple items if they are easy to combine together or create separate issues. For more context about each error please check [a11y report](https://docs.google.com/document/d/1PagfYDwRum_cAcG2NAfS-3m0siVGdgb4Kw6ENaglgNg/edit)

```[tasklist]
### Should have:
- [ ] Correlations: [a11y] Add table headers to td elements
- [ ] Correlations: [a11y] Focus on newly added elements in correlations forms / row expand
- [ ] Correlations: [a11y] Selecting a data source from the picker submits the form
```

```[tasklist]
### Could have:
- [ ] Correlations: [a11y] aria-* attributes do not have valid values in toggle row button
- [ ] Correlations: [a11y] Decide what should be in focus after adding / editing / deleting
```
